Sources/Soto/Extensions/S3/S3+multipart.swift

```diff
@@ -599,6 +633,37 @@ extension S3 {
             throw error
         }
     }
+}
+
+// Internal functions used by multipart upload
+extension S3 {
+    /// Do all the work for opening a file and closing it for MultiUpload function
+    func openFileForMultipartUpload(
+        filename: String,
+        logger: Logger,
+        on eventLoop: EventLoop,
+        threadPoolProvider: ThreadPoolProvider = .createNew,
+        uploadCallback: @escaping (NIOFileHandle, FileRegion, NonBlockingFileIO) async throws -> CompleteMultipartUploadOutput
```

Comment:
Question - if a closure is marked as `async` is it implicitly `@Sendable` because it's crossing a boundary?

Reply:
PS if not this probably wants to be `Sendable`